hl/src/H5LT.c

```diff
@@ -1858,49 +1855,24 @@ H5LTset_attribute_double(hid_t loc_id, const char *obj_name, const char *attr_na
 /*-------------------------------------------------------------------------
  * Function: H5LTfind_attribute
  *
- * Purpose: Inquires if an attribute named attr_name exists attached to
- *          the object loc_id.
- *
- * Programmer: Pedro Vicente
- *
- * Date: May 17, 2006
+ * Purpose:  Checks if an attribute named attr_name exists attached to
+ *           the object loc_id
  *
- * Comments:
- *  Calls the private version of the function
+ * TODO:     Overloading herr_t is not a great idea. This function either
+ *           needs to be rewritten to take a Boolean out parameter in
+ *           HDF5 2.0 or possibly even eliminated entirely as it simply
+ *           wraps H5Aexists.
  *
+ * Return:   An htri_t value cast to herr_t
+ *              Exists:         Positive
+ *              Does not exist: 0
+ *              Error:          Negative
  *-------------------------------------------------------------------------
  */
-
 herr_t
 H5LTfind_attribute(hid_t loc_id, const char *attr_name)
 {
-    return H5LT_find_attribute(loc_id, attr_name);
-}
-
-/*-------------------------------------------------------------------------
- * Function: H5LT_find_attribute
- *
- * Purpose: Inquires if an attribute named attr_name exists attached to the object loc_id.
- *
- * Programmer: Pedro Vicente
- *
- * Date: June 21, 2001
- *
- * Return:
- *  Success: Positive if the attribute exists attached to the
- *              object loc_id. Zero if the attribute does not
- *              exist attached to the object loc_id.
- *
- *  Failure: Negative if something goes wrong within the
- *              library.
- *
- *-------------------------------------------------------------------------
- */
-herr_t
-H5LT_find_attribute(hid_t loc_id, const char *attr_name)
-{
-    htri_t attr_exists = H5Aexists(loc_id, attr_name);
-    return (attr_exists < 0) ? (herr_t)-1 : (attr_exists) ? (herr_t)1 : (herr_t)0;
+    return (herr_t)H5Aexists(loc_id, attr_name);
 }
```

Comment:
This function is completely useless so I just use H5Aexists directly throughout the HL library instead.